src/qgis_geonode/gui/geonode_source_select_provider.py

```diff
@@ -207,23 +207,38 @@ def __init__(self, parent, fl, widgetMode):
         self.spatial_extent_box.setOutputExtentFromCurrent()
         self.spatial_extent_box.setMapCanvas(map_canvas)
 
+        self.restore_settings()
+
+        self.title_le.textChanged.connect(self.save_settings)
+        self.abstract_le.textChanged.connect(self.save_settings)
+        self.keyword_cmb.currentIndexChanged.connect(self.save_settings)
+        self.category_cmb.currentIndexChanged.connect(self.save_settings)
+        self.resource_types_btngrp.buttonToggled.connect(self.save_settings)
+        self.temporal_extent_start_dte.valueChanged.connect(self.save_settings)
+        self.temporal_extent_end_dte.valueChanged.connect(self.save_settings)
+        self.publication_start_dte.valueChanged.connect(self.save_settings)
+        self.publication_end_dte.valueChanged.connect(self.save_settings)
+        self.spatial_extent_box.extentChanged.connect(self.save_settings)
+        self.sort_field_cmb.currentIndexChanged.connect(self.save_settings)
+        self.reverse_order_chb.toggled.connect(self.save_settings)
```

Comment:
Remember to rename the slot here, according to the recommendation below
```suggestion
        self.title_le.textChanged.connect(self.save_search_filters)
        self.abstract_le.textChanged.connect(self.save_search_filters)
        self.keyword_cmb.currentIndexChanged.connect(self.save_search_filters)
        self.category_cmb.currentIndexChanged.connect(self.save_search_filters)
        self.resource_types_btngrp.buttonToggled.connect(self.save_search_filters)
        self.temporal_extent_start_dte.valueChanged.connect(self.save_search_filters)
        self.temporal_extent_end_dte.valueChanged.connect(self.save_search_filters)
        self.publication_start_dte.valueChanged.connect(self.save_search_filters)
        self.publication_end_dte.valueChanged.connect(self.save_search_filters)
        self.spatial_extent_box.extentChanged.connect(self.save_search_filters)
        self.sort_field_cmb.currentIndexChanged.connect(self.save_search_filters)
        self.reverse_order_chb.toggled.connect(self.save_search_filters)
```